src/gui/sdlmain.cpp

```diff
@@ -3478,19 +3504,48 @@ static void GUI_StartUp(Section *sec)
 	startup_state_capslock = keystate & KMOD_CAPS;
 }
 
-static void HandleMouseMotion(SDL_MouseMotionEvent * motion) {
-	if (mouse_is_captured || sdl.mouse.control_choice == Seamless)
-		Mouse_CursorMoved((float)motion->xrel*sdl.mouse.xsensitivity/100.0f,
-						  (float)motion->yrel*sdl.mouse.ysensitivity/100.0f,
-						  (float)(motion->x-sdl.clip.x)/(sdl.clip.w-1)*sdl.mouse.xsensitivity/100.0f,
-						  (float)(motion->y-sdl.clip.y)/(sdl.clip.h-1)*sdl.mouse.ysensitivity/100.0f,
-						  mouse_is_captured);
+
+static void HandleMouseMotion(SDL_MouseMotionEvent *motion)
+{
+	if (motion->xrel < INT16_MIN || motion->xrel > INT16_MAX ||
+	    motion->yrel < INT16_MIN || motion->yrel > INT16_MAX || motion->x < 0 ||
+	    motion->x > UINT16_MAX || motion->y < 0 || motion->y > UINT16_MAX) {
+		LOG_ERR("SDL: Unexpected motion values: rel %dx%d, abs %dx%d",
+		        motion->xrel,
+		        motion->xrel,
+		        motion->x,
+		        motion->y);
+		return;
+	}
```

Comment:
That's 8 pointer de-references and 8 more comparisons per mouse event (more than in the nominal code below); and the mouse event can be running at several hundred Hz.  

I think we'd only expect this in a very broken or bugged version of SDL; can we move it to an assert? If someone has a buggy system, we'll be using debug builds and this will give a nice backtrace into the problem. 

From there, we can then decide if moving this into the release code path (ie: filter junk values) makes sense.

Reply:
This is actually the cost of having mouse data stored in 16-bit variables, not in 32-bit ones (I've changed this after review remark). OK, I'll move this to some kind of assert, everywhere.

Reply:
Thanks @FeralChild64.

Reply:
Done.